src/package-managers/staticRegistry.ts

```diff
@@ -0,0 +1,35 @@
+import fs from 'fs'
+import { GetVersion } from '../types/GetVersion'
+import { Version } from '../types/Version'
+import { Options } from '../types/Options'
+import { StaticRegistry } from '../types/StaticRegistry'
+
+let registry: StaticRegistry = {}
+
+/**
+ * Assigns static registry dependencies to variable given a valid path
+ *
+ * @param path
+ * @returns
+ */
+const initializeRegistry = (path: string | undefined): void => {
+  if (path === undefined || !fs.existsSync(path)) {
+    throw new Error('No or invalid path to static registry was provided. Please try again')
+  }
+  const data = JSON.parse(fs.readFileSync(path, 'utf8'))
+  registry = { ...registry, ...data }
+}
+/**
+ * Fetches the version in static registry.
+ *
+ * @param packageName
+ * @param currentVersion
+ * @param options
+ * @returns A promise that fulfills to string value or null
+ */
+export const latest: GetVersion = (packageName: string, currentVersion: Version, options: Options = {}) => {
+  if (Object.keys(registry).length === 0) {
+    initializeRegistry(options.registry)
+  }
```

Comment:
Caching the registry would break the `--deep` option for monorepos that have different static registries per repo. Perhaps unlikely, but caching bugs are a nightmare to troubleshoot and this is easy to avoid.

I would suggest one of two approaches:

1. Create a separate cache for each value of `options.registry`. i.e. change `registry` to type `{ [key: string]: StaticRegistry }`. This will effectively memoize the `registry` by location.

2. Use `fast-memoize` which is already used by the `npm` adapter to optimize repeat registry access. This effectively hides the implementation details of the cache for the same effect:

    https://github.com/raineorshine/npm-check-updates/blob/d7df1515391ce2084fd73bd4d8b55b09fa57ae1c/src/package-managers/npm.ts#L215-L216

Reply:
Thanks for the tip--I think I'll go with the second option!